atoma-state/src/handlers.rs

```diff
@@ -1105,14 +1105,124 @@ pub(crate) async fn handle_node_key_rotation_event(
         new_public_key,
         tee_remote_attestation_bytes,
     } = event;
+    // TODO: We need to check that the
+    let is_valid =
+        utils::verify_quote_v4_attestation(&tee_remote_attestation_bytes, &new_public_key)
+            .await
+            .is_ok();
     state_manager
         .state
         .update_node_public_key(
             node_id.inner as i64,
             epoch as i64,
             new_public_key,
             tee_remote_attestation_bytes,
+            is_valid,
```

Comment:
The problem I see here, correct me if I'm wrong.
I register TEE node. The node will be selected. And then it rotate to new key which `is_valid` is false. Because the node falsify the key. That way the node can still do the job undetected, but the privacy is gone.
I can do this one time per epoch (current contract conditions it's one time per epoch rotation of key is allowed). It's not much, but still.
Probably once the stack is selected the key should stay the same for the duration of the stack.

Reply:
The x25519 dh public key is not stored in the stack. If you submit an invalid ra, then the proxy don't select you to process those stacks anymore, and you lose the opportunity to accrue value from fees. Moreover, anyone can now start a dispute with the node. That said, I agree this is not the ideal flow, and in time, the quote verification will be peformed on-chain.

Reply:
Yeah. No new stack will be assigned but the old ones lose privacy. In some cases this can be really bad. Maybe in the future we can store the keys on the stack.